pkg/machines/components/nicAdd.jsx

```diff
@@ -0,0 +1,242 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2019 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+import cockpit from 'cockpit';
+import PropTypes from 'prop-types';
+import {
+    Modal,
+    Button
+} from 'patternfly-react';
+
+import { ModalError } from 'cockpit-components-inline-notification.jsx';
+import { getNetworkDevices } from '../helpers.js';
+import { NICBody } from './nicBody.jsx';
+import {
+    getVm
+} from '../actions/provider-actions.js';
+import { attachIface } from '../libvirt-dbus.js';
+
+import './nicEdit.css';
+import 'form-layout.less';
+
+const _ = cockpit.gettext;
+
+const NetworkMacRow = ({ idPrefix, dialogValues, onValueChanged }) => {
+    return (
+        <>
+            <label className='checkbox-inline'>
+                <input id={`${idPrefix}-set-mac`}
+                    type='checkbox'
+                    checked={dialogValues.setNetworkMac}
+                    onChange={e => onValueChanged('setNetworkMac', !dialogValues.setNetworkMac)} />
+                {_("Set MAC address")}
+            </label>
+
+            {dialogValues.setNetworkMac && <>
+                <label className='control-label' htmlFor={`${idPrefix}-mac`}>
+                    {_("Mac Address")}
```

Comment:
I would prefer if this was MAC in upper case, to keep the spelling consistent, and avoid any confusion with the Macintosh computer.